dumpling/export/config.go

```diff
@@ -208,7 +209,7 @@ func (conf *Config) GetDSN(db string) string {
 	// https://github.com/go-sql-driver/mysql#maxallowedpacket
 	hostPort := net.JoinHostPort(conf.Host, strconv.Itoa(conf.Port))
 	dsn := fmt.Sprintf("%s:%s@tcp(%s)/%s?collation=utf8mb4_general_ci&readTimeout=%s&writeTimeout=30s&interpolateParams=true&maxAllowedPacket=0",
-		conf.User, conf.Password, hostPort, db, conf.ReadTimeout)
+		conf.User, conf.Password, hostPort, url.QueryEscape(db), conf.ReadTimeout)
```

Comment:
seems dbName does not support `url.QueryEscape`. So in order to connect to databases like `test!a` I prefer we don't escape it

Reply:
Hi @lance6716 - It must be escaped due to a security vulnerability matter, see the release note of this PR summary. Or do you have a better way to validate the database name?

Reply:
I guess the vulnerability lies in parsing DSN? If so, maybe we can

- use fields of `Config` to set database, read timeout, etc. https://pkg.go.dev/github.com/go-sql-driver/mysql#Config 
- use NewConnector to get a `Connector` from `Config` https://pkg.go.dev/github.com/go-sql-driver/mysql#NewConnector
- use `OpenDB` to get a `DB` from `Connector` https://pkg.go.dev/database/sql#OpenDB

/cc @hawkingrei @gozssky

Reply:
ping @dwisiswant0